samples/gpu/video_writer.cpp

```diff
@@ -31,7 +31,8 @@ int main(int argc, const char* argv[])
     }
 
     cv::cuda::printShortCudaDeviceInfo(cv::cuda::getDevice());
-
+    String outputFileCpu = cv::tempfile(".avi");
+    String outputFileGpu = cv::tempfile(".h264");
```

Comment:
It's sample. User expects to get result in current directory and usually do not know where tmp location. I propose to revert all changes related to `cv::tempfile`